packages/browser/src/breadcrumbs/WebRequestEventSubscriber.ts

```diff
@@ -0,0 +1,70 @@
+import {
+    BreadcrumbLogLevel,
+    BreadcrumbsEventSubscriber,
+    BreadcrumbsManager,
+    BreadcrumbType,
+} from '@backtrace/sdk-core';
+
+export class WebRequestEventSubscriber implements BreadcrumbsEventSubscriber {
+    private _xmlHttpRequestOriginalOpenMethod?: typeof XMLHttpRequest.prototype.open;
+    private _fetchOriginalMethod?: typeof window.fetch;
+
+    public start(breadcrumbsManager: BreadcrumbsManager): void {
+        if ((breadcrumbsManager.breadcrumbsType & BreadcrumbType.Http) !== BreadcrumbType.Http) {
+            return;
+        }
+        const xmlHttpRequestOriginalOpenMethod = XMLHttpRequest.prototype.open;
+        XMLHttpRequest.prototype.open = function (
+            method: string,
+            url: string | URL,
+            async?: boolean,
+            username?: string | null,
+            password?: string | null,
+        ) {
+            breadcrumbsManager.addBreadcrumb(
+                `Sending an HTTP to ${method} request to ${url}.`,
+                BreadcrumbLogLevel.Debug,
+                BreadcrumbType.Http,
+                {
+                    async,
+                },
+            );
+            xmlHttpRequestOriginalOpenMethod.call(this, method, url, async || true, username, password);
+        };
+        this._xmlHttpRequestOriginalOpenMethod = xmlHttpRequestOriginalOpenMethod;
+
+        const fetchOriginalMethod = window.fetch;
+
+        window.fetch = async (...args) => {
+            const [resource, config] = args;
+
+            const response = await fetchOriginalMethod(resource, config);
```

Comment:
Does this throw? It would be nice to add an error breadcrumb.

Reply:
yes it throws and I didn't capture at error to propagate it to the code that wanted to do it, however adding a breadcrumb about the failure is a really nice idea. I will add it